consensus/src/quorum_store/utils.rs

```diff
@@ -251,80 +258,103 @@ impl ProofQueue {
                     .digest_proof
                     .get(digest)
                     .expect("Entry for unexpired digest must exist")
                 {
-                    Some((proof, _)) => {
-                        cur_bytes = cur_bytes + proof.info().num_bytes;
-                        cur_txns = cur_txns + proof.info().num_txns;
+                    Some(proof) => {
+                        cur_bytes += proof.info().num_bytes;
+                        cur_txns += proof.info().num_txns;
                         if cur_bytes > max_bytes || cur_txns > max_txns {
                             // Exceeded the limit for requested bytes or number of transactions.
-                            block_finished = true;
-                            // break;
+                            break;
                         }
                         ret.push(proof.clone());
                     },
                     None => {}, // Proof was already committed, skip.
                 }
             }
-            if *expiration < current_time {
-                if expiration.epoch() < current_time.epoch() {
-                    counters::BATCH_EXPIRED_SMALLER_EPOCH_WHEN_PULL_PROOFS_COUNT.inc();
-                }
+            if *expiration < current_time && !excluded_proofs.contains(digest) {
+                num_expired_but_not_committed += 1;
                 if expiration.round() < current_time.round() {
-                    counters::BATCH_EXPIRED_SMALLER_ROUND_WHEN_PULL_PROOFS_COUNT.inc();
                     counters::GAP_BETWEEN_BATCH_EXPIRATION_AND_CURRENT_ROUND_WHEN_PULL_PROOFS
                         .observe((current_time.round() - expiration.round()) as f64);
                 }
             }
-            // go through the remaining proofs, record the number of unexpired proofs
-            if block_finished {
-                if *expiration >= current_time && !excluded_proofs.contains(digest) {
-                    match self
-                        .digest_proof
-                        .get(digest)
-                        .expect("Entry for unexpired digest must exist")
-                    {
-                        Some((_, local)) => {
-                            remaining_proof_size += 1;
-                            if *local {
-                                remaining_local_proof_size += 1;
-                            }
-                        },
-                        None => {}, // Proof was already committed, skip.
-                    }
+        }
+        counters::EXPIRED_PROOFS_WHEN_PULL.observe(num_expired_but_not_committed as f64);
+        counters::BLOCK_SIZE_WHEN_PULL.observe(cur_txns as f64);
+        counters::BLOCK_BYTES_WHEN_PULL.observe(cur_bytes as f64);
+        counters::PROOF_SIZE_WHEN_PULL.observe(ret.len() as f64);
+
+        ret
+    }
+
+    // returns the number of unexpired local proofs
+    pub(crate) fn clean_local_proofs(&mut self, current_time: LogicalTime) -> usize {
```

Comment:
You return the number of unexpired proofs.
What about the ones that are committed but have not expired yet? 
I think there might be a bug here.